lib/tzinfo/time_or_datetime.rb

```diff
@@ -255,6 +255,29 @@ def -(seconds)
       self + (-seconds)
     end
 
+    # Sets offset from UTC for the TimeOrDateTime. Returns a new TimeOrDateTime,
+    # preserving original class. Does NOT changes values of hour-min-second,
+    # so, (18:00 UTC).offset(3600) will became (18:00 +01:00).
+    # Considers original value's UTC offset wisely.
+    def offset(seconds)
+      return self if seconds == 0
```

Comment:
If the `TimeOrDateTime` represents a `Time` or `DateTime` with an offset, then shouldn't calling `offset(0)` change the offset to 0 instead of leaving it unchanged?